File: app/utils/response.py
from typing import Any, Optional, Dict, List, Union

def success_response(data: Union[List[Dict[str, Any]], int]) -> Dict[str, Any]:
    """
    成功响应格式化
    
    Args:
        data: 查询结果列表或受影响的行数
        
    Returns:
        统一格式的响应字典
    """
    return {
        "errCode": 0,
        "data": data,
        "errMsg": None
    }

def error_response(error_code: int, error_message: str) -> Dict[str, Any]:
    """
    错误响应格式化
    
    Args:
        error_code: 错误代码
        error_message: 错误信息
        
    Returns:
        统一格式的响应字典
    """
    return {
        "errCode": error_code,
        "data": None,
        "errMsg": error_message
    } 

File: utils/response.go
package utils

import (
	"github.com/gin-gonic/gin"
)

// Response 定义统一的响应结构
type Response struct {
	ErrCode int         `json:"errCode"`
	Data    interface{} `json:"data"`
	ErrMsg  interface{} `json:"errMsg"`
}

// SuccessResponse 成功响应
func SuccessResponse(data interface{}) Response {
	return Response{
		ErrCode: 0,
		Data:    data,
		ErrMsg:  nil,
	}
}

// ErrorResponse 错误响应
func ErrorResponse(errorCode int, errorMessage string) Response {
	return Response{
		ErrCode: errorCode,
		Data:    nil,
		ErrMsg:  errorMessage,
	}
}

// ResponseSuccess 返回成功响应
func ResponseSuccess(c *gin.Context, data interface{}) {
	c.JSON(200, SuccessResponse(data))
}

// ResponseError 返回错误响应
func ResponseError(c *gin.Context, httpCode, errorCode int, errorMessage string) {
	c.JSON(httpCode, ErrorResponse(errorCode, errorMessage))
} 